aot-plugin/src/main/java/io/micronaut/gradle/aot/MicronautAotPlugin.java

```diff
@@ -320,18 +320,15 @@ private void registerJavaExecOptimizedRun(Project project,
                 // https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/385
                 task.getOutputs().upToDateWhen(t -> false);
                 task.setClasspath(project.files(jarTask, optimizedRuntimeClasspath));
-                task.doFirst(new Action<Task>() {
-                    @Override
-                    public void execute(Task t) {
-                        if (task.getLogger().isDebugEnabled()) {
-                            task.getLogger().debug(
-                                    "Running optimized entry point: " + task.getMainClass().get() +
-                                    "\nClasspath:\n    " + task.getClasspath().getFiles()
-                                            .stream()
-                                            .map(File::getName)
-                                            .collect(Collectors.joining("\n    "))
-                            );
-                        }
+                task.doFirst((t) -> {
```

Comment:
As far as I'm aware, you'll need to revert this change: lambdas are not compatible with the configuration cache.

Reply:
fixed